src/plugins/discover/public/application/components/sidebar/discover_index_pattern_management.test.tsx

```diff
@@ -59,15 +64,51 @@ describe('Discover IndexPattern Management', () => {
 
   const editField = jest.fn();
 
-  test('renders correctly', () => {
-    const component = mountWithIntl(
+  const mountComponent = () => {
+    return mountWithIntl(
       <DiscoverIndexPatternManagement
         services={mockServices}
         editField={editField}
         selectedIndexPattern={indexPattern}
         useNewFieldsApi={true}
       />
     );
+  };
+
+  test('renders correctly', () => {
+    const component = mountComponent();
     expect(component).toMatchSnapshot();
+    expect(component.find(EuiPopover).length).toBe(1);
```

Comment:
I'd recommend additionally testing for the existence of the button, and removing the snapshot, since then it doesn't have to be regenerated if underlying components oder dependencies change

Reply:
I like the snapshot for that very reason - it quickly points to an underlying change in the component structure.